streams/src/main/java/org/apache/kafka/streams/processor/internals/StateManagerUtil.java

```diff
@@ -104,17 +104,16 @@ static void closeStateManager(final Logger log,
             if (stateDirectory.lock(id)) {
                 try {
                     stateMgr.close();
-
+                } catch (final ProcessorStateException e) {
+                    firstException.compareAndSet(null, e);
+                } finally {
                     if (wipeStateStore) {
                         log.debug("Wiping state stores for {} task {}", taskType, id);
                         // we can just delete the whole dir of the task, including the state store images and the checkpoint files,
                         // and then we write an empty checkpoint file indicating that the previous close is graceful and we just
                         // need to re-bootstrap the restoration from the beginning
                         Utils.delete(stateMgr.baseDir());
```

Comment:
Seems weird to skip deleting the dirty state on a dirty close just because we caught an exception while closing the state (probably in most cases a dirty close would mean something would throw, for example calling `flush` when we've dropped out of the group)

Reply:
Well, it could be some other issue and we would wipe the state on "resuming" the task anyway. Not sure if there are other things to consider? \cc @guozhangwang

Reply:
Right, it's not a correctness issue but it's additional needless overhead to go through the whole cycle of initializing a task, getting a TaskCorrupted, wiping it then, and finally restarting it. Of course if we keep hitting an issue during `closeDirty` then we might never wipe the state, which does seem like a real problem. For example if there's some issue with the state, like the files are actually corrupted or something